Fix some current open issues

mattsta opened the pull request:
Reviews / checks welcome.

I ran this against live API usage and the new tick fields appear to work without crashing anything so far (though, I can't get all of them to populate correctly because the server-side tick mappings aren't the client-request tick mappings, and I don't see a true client tick map population list anywhere).

Will update/integrate these refactors if no bad reports come back after a couple days.

mattsta committed:
Implement missing tick types

Also refactors some handling to use the more modern O(1) setting
patterns.

Adds missing tick types present in the current IBKR ibapi client we
hadn't updated internally in a while.

Currently untested, but looks good so far. Some specific data types or
data extraction formats may need to be adjusted when we test against
live data.

Fixes #182
Fixes #182

mattsta committed:
Add explicit `tzdata` dependency

Also removes obsolete import guard since Python 3.9+ guarantees zoneinfo
exists.

Fixes #188

mattsta committed:
Improve event loop fetching

This wasn't a global problem and only triggered edge cases for people
developing with too much (unnecessary?) complexity, but now we don't
cache the event loop and instead request the live event loop each time.

This is technically about 10x slower than using a cached event loop
reader, but the difference is just +20ns slower if we ask python to
lookup the active event loop each time vs using it cached on first
access forever.

Fixes #160
Fixes #186
Fixes #159

mattsta committed:
Use more modern python syntax

We are currently Python 3.10 minimum, so use Python 3.10 syntax
everywhere.

mattsta committed:
Improve ruff check warnings

Lots of warnings are just from notebooks which we can ignore due to
import issues. Others are smaller things we fixed in-place.

gnzsnz commented:
i feel your pain, this is my attempt at having a ticker tick mapping

https://github.com/gnzsnz/ib_async/blob/c6787eade7084bcb3376ace6ac8ac89f32993a27/ib_async/ticker.py#L37

it's 100% somewhat complete.

Edit: I forgot the actual TickType mapping https://github.com/gnzsnz/ib_async/blob/c6787eade7084bcb3376ace6ac8ac89f32993a27/ib_async/objects.py#L140

Edit II: and the tick types document https://docs.google.com/spreadsheets/d/e/2PACX-1vTJ8av3r1DnW0g9UeOI0ASFr6LDj4ct-Migz3sC4grOnBGuHp_dkq8Djg2h0-uQ_MaBlkgI3a5V_eNY/pubhtml#gid=0

"Ticker" tab needs to be aligned with your changes. "Group by delivery method" maps message id with tick id, which needs to be aligned with the ticker.py mappings

gnzsnz commented:
do you plan to include a fix for #172 on this PR?

gnzsnz committed:
Fix Timebars typo

This typo is breaking timebars class. there is no `isUnset` method in timebars class. reverting change to `isNan` helper function.

Fixes #197

gnzsnz committed:
Fix setuptools deprecation warning

Fixes #198

gnzsnz committed:
Fix flexReport URL

Provides mechanism to overwrite default ULR with an environment variable

Fixes #199
Fixes #172

mattsta committed:
New release: 2.1.0

Some fixes to community complaints about code organization or
assumptions around caching and system use edge cases etc.

See changelog or git log for full details.

Also includes starting to double up some pyproject.toml config for
eventually converting this from a poetry project to a uv project.

mattsta committed:
Update CI build paramaters

Add python 3.14 and use configured poetry config for testing
